editor/plugins/animation_library_editor.cpp

```diff
@@ -0,0 +1,689 @@
+				undo_redo->add_undo_method(this, "_update_editor", player);
+				updating = true;
+				undo_redo->commit_action();
+				updating = false;
+				ti->set_metadata(0, text);
+				if (text == "") {
+					ti->set_suffix(0, TTR("[Global]"));
+				} else {
+					ti->set_suffix(0, "");
+				}
+			}
+		} else {
+			// Renamed anim
+			StringName library = ti->get_parent()->get_metadata(0);
+			Ref<AnimationLibrary> al = player->call("get_animation_library", library);
+
+			if (al.is_valid()) {
+				if (al->has_animation(text)) {
+					restore_text = true;
+				} else {
+					undo_redo->create_action(vformat(TTR("Rename Animation : %s"), text));
+					undo_redo->add_do_method(al.ptr(), "rename_animation", old_text, text);
+					undo_redo->add_undo_method(al.ptr(), "rename_animation", text, old_text);
+					undo_redo->add_do_method(this, "_update_editor", player);
+					undo_redo->add_undo_method(this, "_update_editor", player);
+					updating = true;
+					undo_redo->commit_action();
+					updating = false;
+
+					ti->set_metadata(0, text);
+				}
+			} else {
+				restore_text = true;
+			}
+		}
+	}
+
+	if (restore_text) {
+		ti->set_text(0, old_text);
+	}
+}
+
+void AnimationLibraryEditor::_button_pressed(TreeItem *p_item, int p_column, int p_button) {
+	if (p_item->get_parent() == tree->get_root()) {
+		// Library
+		StringName lib_name = p_item->get_metadata(0);
+		Ref<AnimationLibrary> al = player->call("get_animation_library", lib_name);
+		switch (p_button) {
+			case LIB_BUTTON_ADD: {
+				add_library_dialog->set_title(TTR("Animation Name:"));
+				add_library_name->set_text("");
+				add_library_dialog->popup_centered();
+				add_library_name->grab_focus();
+				adding_animation = true;
+				adding_animation_to_library = p_item->get_metadata(0);
+				_add_library_validate("");
+			} break;
+			case LIB_BUTTON_LOAD: {
+				adding_animation_to_library = p_item->get_metadata(0);
+				List<String> extensions;
+				ResourceLoader::get_recognized_extensions_for_type("Animation", &extensions);
+
+				file_dialog->clear_filters();
+				for (const String &K : extensions) {
+					file_dialog->add_filter("*." + K);
+				}
+
+				file_dialog->set_title(TTR("Load Animation"));
+				file_dialog->set_file_mode(EditorFileDialog::FILE_MODE_OPEN_FILE);
+				file_dialog->set_current_file("");
+				file_dialog->popup_centered_ratio();
+
+				file_dialog_action = FILE_DIALOG_ACTION_OPEN_ANIMATION;
+
+			} break;
+			case LIB_BUTTON_PASTE: {
+				Ref<Animation> anim = EditorSettings::get_singleton()->get_resource_clipboard();
+				if (!anim.is_valid()) {
+					error_dialog->set_text(TTR("No animation resource in clipboard!"));
+					error_dialog->popup_centered();
+					return;
+				}
+
+				anim = anim->duplicate(); // Users simply dont care about referencing, so making a copy works better here.
+
+				String base_name;
+				if (anim->get_name() != "") {
+					base_name = anim->get_name();
+				} else {
+					base_name = "Pasted Animation";
```

Comment:
Could be localized too I guess?
```suggestion
					base_name = TTR("Pasted Animation");
```